src/storage/src/hummock/compactor/mod.rs

```diff
@@ -782,3 +784,74 @@ fn build_multi_compaction_filter(compact_task: &CompactTask) -> MultiCompactionF
 
     multi_filter
 }
+
+async fn generate_splits(compact_task: &mut CompactTask, context: Arc<Context>) {
+    let sstable_infos = compact_task
+        .input_ssts
+        .iter()
+        .flat_map(|level| level.table_infos.iter())
+        .collect_vec();
+
+    let compaction_size = compact_task
+        .input_ssts
+        .iter()
+        .flat_map(|level| level.table_infos.iter())
+        .map(|table_info| table_info.file_size)
+        .sum::<u64>();
+
+    let sstable_size = (context.options.sstable_size_mb as u64) << 20;
+    if compaction_size > sstable_size {
+        let mut indexes = vec![];
+        // preload the meta and get the smallest key to split sub_compaction
+        for sstable_info in sstable_infos {
+            indexes.extend(
+                context
+                    .sstable_store
+                    .sstable(sstable_info, &mut StoreLocalStatistic::default())
+                    .await
+                    .unwrap()
+                    .value()
+                    .meta
+                    .block_metas
+                    .iter()
+                    .map(|block| {
+                        let data_size = block.len;
+                        let full_key =
+                            FullKey::from_user_key_slice(user_key(&block.smallest_key), u64::MAX)
+                                .into_inner();
+                        (data_size, full_key.to_vec())
+                    })
+                    .collect_vec(),
+            );
+        }
+        // sort_by key, as for every data block has the same size;
+        indexes.sort_by(|a, b| VersionedComparator::compare_key(a.1.as_ref(), b.1.as_ref()));
+        let mut splits: Vec<KeyRange_vec> = vec![];
+        splits.push(KeyRange_vec::new(vec![], vec![]));
+        let parallelism = std::cmp::min(
+            indexes.len() as u64,
+            context.options.max_sub_compaction as u64,
+        );
+        let sub_compaction_data_size = if compaction_size > sstable_size && parallelism > 1 {
```

Comment:
As @Li0k  said. It is not necessary.
You can refactor it as this:
```rust
let sub_compaction_data_size = compaction_size / parallelism;
```

Because `sub_compaction_data_size` would never be used if `parallelism > 1` and `compaction_size > sstable_size` do not meet.